pkg/apis/provisioning/v0alpha1/types.go

```diff
@@ -56,50 +45,43 @@ type GitHubRepository struct {
 	GenerateDashboardPreviews bool `json:"generateDashboardPreviews,omitempty"`
 }
 
-func (r *GitHubRepository) IsEmpty() bool {
-	// we don't need to check options here, just the most important stuff to actually connect.
-	return r == nil || r.Owner == "" || r.Repository == ""
-}
+// RepositoryType defines the types of Repository
+// +enum
```

Comment:
for my own learning's sake: what tool depends on this tag? openapi-gen?

Reply:
kube-openapi looks for it https://github.com/kubernetes/kube-openapi/blob/master/pkg/generators/enum.go#L168

and then will set the enum values in openapi:
https://github.com/grafana/grafana/pull/96438/files#diff-2d0d891a8c382777f4d12dd64bb2c91b5cec99cf3245631e99f4b4f6c560e12dR238

i don't think that is yet supported in the app-platform cue > CRD codegen